### PR TITLE
logic add to cart

### PR DESCRIPTION
**Pull request by Baium13**
creating home, cart, and checkout pages. Displaying data about products, etc. on all pages, creating logic for selecting a product to cart

### CODE REVIEW

File: shop/models.py [hunk 1]
```diff
@@ -82,6 +86,36 @@ def __str__(self):
...
+    product = models.ForeignKey(Product, on_delete=models.SET_NULL, null=True)
+    quantity = models.PositiveIntegerField(default=0, null=True, blank=True)
+    line_price = models.DecimalField(max_digits=10, decimal_places=2, null=False)
+    sku = models.CharField(max_length=50)
+    upc = models.CharField(max_length=50)
+    description = models.CharField(max_length=255, null=True)
+
+    @property
+    def get_price(self):
+        product = ProductPrice.objects.filter(product=self.product)
```
Comment (artemijan) [inline shop/models.py:100]: what if product is removed from the system? OrderLine shouldn't rely on product

File: shop/models.py [hunk 2]
```diff
@@ -82,6 +86,36 @@ def __str__(self):
...
+        price = product[0].price
+        return price
+
+    @property
+    def get_currency(self):
+        product = ProductPrice.objects.filter(product=self.product)
```
Comment (artemijan) [inline shop/models.py:106]: the same as above

File: shop/models.py [hunk 3]
```diff
@@ -44,25 +48,25 @@ class Order(AbstractAuditableModel):
...
-    quantity = models.PositiveIntegerField()
-    line_price = models.DecimalField(max_digits=10, decimal_places=2, null=False)
-    sku = models.CharField(max_length=50)
-    upc = models.CharField(max_length=50)
-    description = models.CharField(max_length=255, null=True)
-
-    def __str__(self):
-        return self.description
+    @property
+    def get_all_item(self):
```
Comment (artemijan) [inline shop/models.py:66]: what this property does? it's not correct from what I can see

File: shop/models.py [hunk 4]
```diff
@@ -44,25 +48,25 @@ class Order(AbstractAuditableModel):
...
-    total = models.DecimalField("Order total", max_digits=10, decimal_places=2)
+    total = models.DecimalField("Order total", max_digits=10, decimal_places=2, null=False)
     currency = models.CharField(max_length=3, choices=Currency.choices())
     shipping_date = models.DateField()
 
     def __str__(self):
         return self.number
 
+    def save(self, *args, **kwargs):
+        all_price = OrderLine.objects.filter(order=self)
```
Comment (artemijan) [inline shop/models.py:59]: you can use aggregation functions

**Comment by Baium13** [comment]
Old Version

**Pull request closed without merging**